### ISSUE TITLE
GPU Specs

### ISSUE THREAD

**Issue by partha2409** [funcwj/conv-tasnet#1]
Hi, Thanks for the repo! what is your GPU specs. I am not able to fit 32 or 20 batches in 2 GPUs. I use a TitanX 12 GB and Tesla k40 12GB.  Also can you share your loss curve?

**Comment by funcwj** [comment]
I ran experience on Tesla P40 (24GB). For BN I used 32 batches and ran on 2 gpus. Seems you need to reduce your batch size.

**Comment by funcwj** [comment]
For BN, loss for first 30 epochs looks like:
```
Epoch  1: train = -3.6581 | dev = -5.9427 
Epoch  2: train = -7.9997 | dev = -9.2260 
Epoch  3: train = -9.8297 | dev = -9.7800 
Epoch  4: train = -10.7609 | dev = -10.9702 
Epoch  5: train = -11.3718 | dev = -11.5721 
Epoch  6: train = -11.8733 | dev = -11.9896 
Epoch  7: train = -12.2721 | dev = -12.2956 
Epoch  8: train = -12.5596 | dev = -6.0468
Epoch  9: train = -12.9008 | dev = -12.8068 
Epoch 10: train = -13.1976 | dev = -12.9763 
Epoch 11: train = -13.4194 | dev = -12.9821 
Epoch 12: train = -13.6278 | dev = -13.2990 
Epoch 13: train = -13.8484 | dev = -13.4017 
Epoch 14: train = -14.0084 | dev = -13.7790 
Epoch 15: train = -14.1521 | dev = -13.7372
Epoch 16: train = -14.3159 | dev = -13.6879
Epoch 17: train = -14.4198 | dev = -13.5005
Epoch 18: train = -14.5691 | dev = -13.3061
Epoch 19: train = -15.1811 | dev = -14.5349 
Epoch 20: train = -15.4449 | dev = -14.0975
Epoch 21: train = -15.5488 | dev = -14.5017
Epoch 22: train = -15.6797 | dev = -14.5178
Epoch 23: train = -15.7873 | dev = -13.9141
Epoch 24: train = -16.1401 | dev = -14.8336 
Epoch 25: train = -16.2666 | dev = -14.8130
Epoch 26: train = -16.3392 | dev = -14.8859 
Epoch 27: train = -16.4343 | dev = -14.9917 
Epoch 28: train = -16.5049 | dev = -15.0309 
Epoch 29: train = -16.5485 | dev = -14.9959
Epoch 30: train = -16.5969 | dev = -14.7695
```

**Comment by partha2409** [comment]
Great Thanks ! one more clarification. I'm using my own dataloader and trying to use your loss function. In your 'trainer.py' you divide the th.sum(max_perutt) by 'N' in the last line. Is this 'N' - number of batches or Number of Speakers?  TIA

**Comment by funcwj** [comment]
N means the number of chunks in current mini-batch

**Comment by partha2409** [comment]
Which is basically the batch_Size right? or am i misinterpreting something?

**Comment by funcwj** [comment]
In most time, N equals to batch size, this is determined by how you design the dataloader.

**Comment by funcwj** [comment]
In this experience, I prefer to choose the batch size which could fill the memory of gpus, egs 32 for 2 devices, 48GB in total

**Comment by partha2409** [comment]
ok thanks a lot :) i will try it out
